core/src/main/java/org/elasticsearch/indices/breaker/StaticCircuitBreakerService.java

```diff
@@ -26,14 +26,30 @@
 /**
  * Class that returns a breaker that never breaks
  */
-public class NoneCircuitBreakerService extends CircuitBreakerService {
+public class StaticCircuitBreakerService extends CircuitBreakerService {
```

Comment:
I *think* I understand why this is needed - in a test you needed to have this based on a custom breaker for the new inflight breaker type. I think this signals a bigger problem. This class currently returns the same breaker for all types. Which used to be fine (for all but stats reporting, which was hard coded to field data), because the logic was "return me something that does nothing anyway". Now it becomes "return me the same breaker regardless of what I asked for". 

I think the right solution is to have the set of breakers predefine and registered on the circuit breaker module and have all the breakers configured to NoopCircuitBreaker . That is a big change.  

Alternatively we can make this class be `SharedCircuitBreakerService` where always pass a single circuit breaker in it's constructor and it's document that all requests (including stats) goes to that one.  We can then instantiate it with the NoopCircuitBreaker in CircuitBreakerModule based on config.